doc/api/metrics.md

```diff
@@ -0,0 +1,1040 @@
+function handleRequest(req, res) {
+  const timer = requestTimer.create({ endpoint: req.url });
+
+  apiCalls.increment();
+
+  // Process request...
+
+  timer.stop();
+}
+```
+
+## Metric Types
+
+### `metrics.counter(name[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the counter metric.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.Counter}
+
+Creates a counter metric that tracks cumulative values.
+
+```mjs
+import { counter } from 'node:metrics';
+
+const errorCount = counter('errors.total', { component: 'database' });
+
+errorCount.increment();     // Increment by 1
+errorCount.increment(5);    // Increment by 5
+errorCount.decrement(2);    // Decrement by 2
+```
+
+
+### `metrics.gauge(name[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the gauge metric.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.Gauge}
+
+Creates a gauge metric that represents a single value at a point in time.
+
+```mjs
+import { gauge } from 'node:metrics';
+import { memoryUsage } from 'node:process';
+
+const memory = gauge('memory.usage.bytes');
+
+memory.reset(memoryUsage().heapUsed);
+memory.applyDelta(1024);  // Add 1024 to current value
+```
+
+### `metrics.meter(name, interval[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the meter metric.
+* `interval` {number} The time window in milliseconds for rate calculation.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.Meter}
+
+Creates a meter metric that measures the rate of events over time.
+
+```mjs
+import { meter } from 'node:metrics';
+
+const requestRate = meter('requests.rate', 60000); // 1 minute window
+
+requestRate.mark();     // Mark one event
+requestRate.mark(10);   // Mark 10 events
+```
+
+### `metrics.timer(name[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the timer metric.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.TimerFactory}
+
+Creates a timer factory for measuring durations.
+
+```mjs
+import { timer } from 'node:metrics';
+
+const dbQueryTimer = timer('db.query.duration');
+
+const t = dbQueryTimer.create({ query: 'SELECT * FROM users' });
+// Perform database query...
+const duration = t.stop(); // Returns duration in milliseconds
+```
+
+### `metrics.uniqueSet(name[, meta])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the unique set metric.
+* `meta` {Object} Optional metadata to attach to all reports.
+* Returns: {metrics.UniqueSet}
+
+Creates a unique set metric which counts distinct objects. Uniqueness is
+determined through a `WeakSet`, so it follows the same identity rules.
+
+```mjs
+import { uniqueSet } from 'node:metrics';
+
+const uniqueUsers = uniqueSet('users.unique');
+
+uniqueUsers.add(user);        // Only counted once per unique value
```

Comment:
How useful it is to have a `uniqueSet` that's based on the object identity? For two distinct requests, a single user object will be two distinct objects.

Reply:
Not sure. That's not the best example though. There's certainly scenarios where object identity could be useful for distinctness.

I mainly just wanted to try the WeakMap approach because just stuffing everything into one giant set forever sounds like a leak to me. 😅

Reply:
Could maybe try a Set + WeakSet pair with only numeric types going to the regular Set. It'd be at least _less_ of a leak that way. 🤔 

Also could impose some time window on it and delete old values, even though statsd doesn't appear to do that.

Reply:
Perhaps the `uniqueSet` could accept as an option a callback for overriding the uniqueness determination:

```js
const uniqueUser = createUniqueSet('users.unique', {
  equivalent(a, b) {
    return a.id === b.id;
  }
});

uniqueUser.add({ id: 1 });
uniqueUser.add({ id: 2 });
uniqueUser.add({ id: 1 });
```

Reply:
I opted for removing unique set entirely and am instead aiming for value producer wrapper types. I've got Timer now producing values to send to either a gauge or a counter, I'll likely leave it up to ecosystem to provide wrappers for tracking uniqueness counts. The _pattern_ to provide such value producers is in place in any case, though will probably get further polish as I continue to iterate.